docs/developers/api-basics.md

```diff
@@ -0,0 +1,116 @@
+
+Create a new class (let's say `com.example.velocityplugin.VelocityTest`) and paste this in:
+
+```java
+package com.example.velocityplugin;
+
+import com.google.inject.Inject;
+import com.velocitypowered.api.plugin.Plugin;
+import com.velocitypowered.api.proxy.ProxyServer;
+import org.slf4j.Logger;
+
+@Plugin(id = "myfirstplugin", name = "My First Plugin", version = "0.1.0-SNAPSHOT",
+        description = "I did it!", authors = {"Me"})
+public class VelocityTest {
+
+    private final ProxyServer server;
+    private final Logger logger;
+
+    @Inject
+    public VelocityTest(ProxyServer server, Logger logger) {
+        this.server = server;
+        this.logger = logger;
+
+        logger.info("Hello there! I made my first plugin with Velocity.");
+    }
+}
+```
+
+What did you just do there? There's quite a bit to unpack, so let's focus on the Velocity-specific bits:
+
+```
+@Plugin(id = "myfirstplugin", name = "My First Plugin", version = "0.1.0-SNAPSHOT",
+        description = "I did it!", authors = {"Me"})
+public class VelocityTest {
+```
+
+This tells Velocity that this class contains your plugin (myfirstplugin) so that it can be loaded once the proxy starts up.
+Velocity will detect where the plugin will reside when you compile your plugin.
+
+Moving on, what's this?
+
+```java
+@Inject
+public VelocityTest(ProxyServer server, Logger logger) {
+    this.server = server;
+    this.logger = logger;
+
+    logger.info("Hello there, it's a test plugin I made!");
+}
+```
+
+This looks like magic! How is Velocity doing this? The answer lies in the `@Inject`, which indicates that Velocity should
+inject a ProxyServer and the Logger when constructing your plugin. These two interfaces will help you out as you begin
+working with Velocity. We won't talk too much about dependency injection: all you need to know is that Velocity will
+do this.
+
+All you need to do is build your plugin, put it in your `plugins/` directory, and try it! Isn't that nice? In the next
+section you'll learn more about how to use the API.
+
+### Choosing @Plugin Information
+
+Choose your plugin's ID wisely. Other plugins will use this ID to depend on yours. If you change it, you could break compatibility.
+
+The plugin name is somewhat less important. It will be shown to users as the display name of your plugin, but tweaking it will not be catastrophic.
+
+For the version, we recommend sticking to semantic versioning - you can read more about this concept at [semver.org](https://semver.org/). Basically, use 3 numbers in your version, such as 1.4.25. Increment the major number when you make a backwards-incompatible breaking change, increment the minor number when you add functionality in a compatible manner, and increment the patch number when you fix a bug or make an otherwise unnoticeable change in the implementation.
+
+You can also describe your plugin's URL, authors, and description in your `@Plugin` annotation. We'll get to plugin dependencies later.
```

Comment:
We could potentially include the URL into the example plugin annotation that way the basics are covered

Reply:
Good idea, since the authors and description are already included.